docs/madr/decisions/031-locality-aware-load-balancing.md

```diff
@@ -0,0 +1,479 @@
+          - from:                                           # (6)
+              zones: ["zone-1", "zone-2"]
+            to:                                             # (7)
+              type: Only                                    # (8)
+              zones: ["zone-1", "zone-2"]
+          - from:
+              zones: ["zone-3"]
+            to:
+              type: AnyExcept
+              zones: ["zone-1"]
+          - to:
+              type: Any
+      failoverThreshold: 
+        percentage: 70                                      # (7)
+```
+
+
+(1) In `localZone` section, you configure the affinity of tags used for load balancing. If list is empty or localZone is not specified, load balancing works normally, without taking any priorities into account. 
+
+(2) In `key` field, you define the name of the tag.
+
+(3) In `weight` field, you define proportional load that the group should take. Default is calculated based on the order and might and first in the list is going to have the highest weight while the last the lowest.
+
+Example:
+```yaml
+affinityTags:
+  - key: "k8s.io/node"
+  - key: k8s.io/az"
+  - key: k8s.io/region"
+```
+
+Would create weights:
+
+```yaml
+affinityTags:
+  - key: "k8s.io/node"
+    weight: 900
+  - key: k8s.io/az"
+    weight: 90
+  - key: k8s.io/region"
+    weight: 9
+```
+
+and all other endpoints are going to have the weight of 1. When configuring weight you need to either specify weights 
+for all tags or don't specify weights at all.
+
+(4) In `crossZone` section, you configure zones priority.
+
+(5) In `failover` section, you configure list of zone priorities, first rules will have the highest load balancing priority and last in order will have the lowest priority.
+
+(6) In `from` section, you configure to which zones this rule applies. This rule will apply to all DPPs in `zone-1` and `zone-2`. If not defined is for all zones.
+
+(7) In `to` section, you configure to which zones traffic should be redirected. This field is required.
+
+(8) `type` field lets you control how fallback zones are picked. Allowed values:
+- `Only` - will load balance traffic only to zones listed in `zones` field 
+- `None` - will not load balance traffic any further
+- `Any` - will load balance traffic to all zones
+- `AnyExcept` - will load balance traffic to all zones except ones specified in `zones` field
+
+(9) In `failoverThreshold` you configure the percentage of live endpoints below which we will start load balancing to the 
+next priority. This is the inverse of [Envoy overprovisioning factor](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/load_balancing/overprovisioning#arch-overview-load-balancing-overprovisioning-factor). 
+Default: 50%
+
+
+### API examples based on use cases
+
+#### Keeping traffic in same zone (disable cross zone traffic)
+
+This:
+```yaml
+targetRef:
+  kind: Mesh
+to:
+  targetRef:
+    kind: MeshService
+    name: backend
+  defaults: {}
```

Comment:
I'd rather disallow `default: {}` because the result will be whatever is specified in top-level policies after merging. So it's the same as not having the `targetRef` for the `backend` service at all. 

On the other hand, having

```yaml
defaults:
  localityAwareness:
    localZone: 
      affinityTags: []
```

makes much more sense because you're actually overriding the behaviour of the less specific policies

Reply:
Yes, it makes sense. We are going to change it.